app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,9 @@
+class LikesController < ApplicationController
+  def create
+    @like = current_user.like.new
+    @like.post_id = params[:post_id]
+    @like.save
+    flash[:success] = 'Liked sucessfully!'
+    redirect_to "/users/#{current_user.id}/posts"
+  end
```

Comment:
- [ ]  consider exploring` find method ` and retrieve the post that user will add like to, in this way : ``@post = Post.find(params[:post_id])``, and have access to the post, will give access to its id `post.id` form the db, and then you will be able to access the like, refering to the current user like the way you did, and pass the post id. ``@like = @current_user.likes.new(post_id: @post.id)`` and then use respond_to method to save your like like you did it in post controllers

Reply:
Resolved